default/AI/freeorion_debug/option_tools.py

```diff
@@ -42,58 +45,63 @@ def get_sectioned_option_dict():
     Return options for AI
     :return: ordered dict of ordered dicts of options
     """
-    
+
     if not sectioned_options:
         _parse_options()
     return sectioned_options
 
 
 def _parse_options():
     # get defaults; check if don't already exist and can write
-    default_file = _get_default_file_path()
+    default_file = _get_option_file()
     if os.path.exists(default_file):
         config = SafeConfigParser()
         config.read([default_file])
     else:
-        if platform.system() != "Linux":
-            default_file = ""
         try:
             config = _create_default_config_file(default_file)
         except IOError:
-            print "AI Config: default file is not present and not writable at location %s" % default_file
-            config = _create_default_config_file("")
+            print ("AI Config: default file is not present and not writable at location %s\n" % default_file)
+            config = _create_default_config_file(os.path.join(fo.getUserDir(), CONFIG_DEFAULT_FILE))
 
-    option_path = _get_option_file_path()
+    option_file = _get_option_file()
     # if not os.path.exists(option_path):
     #    raise Exception('Error, option path "%s" does not exists.' % option_path)
 
     # read the defaults and then the specified config path
-    if option_path:
-        config_files = [option_path]
+    if option_file:
+        config_files = [option_file]
         configs_read = config.read(config_files)
-        print "AI Config read config file(s): %s" % configs_read
+        print ("AI Config read config file(s): %s" % configs_read)
         if len(configs_read) != len(config_files):
-            print "AI Config Error; could NOT read config file(s): %s" % list(set(config_files).difference(configs_read))
+            print ("AI Config Error; could NOT read config file(s): %s"
+                   % list(set(config_files).difference(configs_read)))
     for section in config.sections():
         sectioned_options.setdefault(section, odict())
         for k, v in config.items(section):
             flat_options[k] = v
             sectioned_options[section][k] = v
 
 
-def _get_option_file_path():
+def _get_option_file():
     # hack to allow lunch this code separately to dump default config
-    ai_path = fo.getAIDir()
-    option_file = fo.getAIConfigStr()
+    ai_path = _get_default_file_path()
+    option_file = CONFIG_DEFAULT_FILE
     if ai_path and option_file:
         return os.path.join(ai_path, option_file)
     else:
         return None
 
 
 def _get_default_file_path():
-    # TODO: determine more robust treatment in case ResourceDir is not writable by user
-    return os.path.join(fo.getAIDir() or ".", DEFAULT_CONFIG_FILE)
+    try:
+        if os.path.isdir(fo.getUserDir()) and not os.path.isdir(CONFIG_DEFAULT_DIR):
+            os.makedirs(CONFIG_DEFAULT_DIR)
+    except OSError:
+        print ("AI Config Error: could not create path %s" % CONFIG_DEFAULT_DIR)
```

Comment:
Don't print errors. Errors should got to stderr. In that case it will be easy to notice them in logs.